dglc/dglc_datamode_noevolve_mod.F90

```diff
@@ -361,31 +370,105 @@ subroutine dglc_datamode_noevolve_advance(pio_subsystem, io_type, io_format, &
          do ns = 1,num_icesheets
-            lsize = size(Fgrg_rofi(ns)%ptr)
+
+            ! Get mesh info
+            call ESMF_MeshGet(model_meshes(ns), elementdistGrid=distGrid, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+            allocate(ice_runoff_out(lsize))
+            ice_runoff_out(:) = 0.d0
+            loc_pos_smb(1) = 0.d0
+            Tot_pos_smb(1) = 0.d0
+            loc_neg_smb(1) = 0.d0
+            Tot_neg_smb(1) = 0.d0
+            num_Tot = 0.d0
+            rat = 0.d0
+
+            ! For No Evolve to reduce negative ice fluxes from DGLC, we will
+            ! Calculate the total positive and total negative fluxes on each
+            ! processor first (local totals).
             do ng = 1,lsize
-               if (is_in_active_grid(usrf(ng))) then
-                  Fgrg_rofi(ns)%ptr(ng) = Flgl_qice(ns)%ptr(ng)
-               else
-                  Fgrg_rofi(ns)%ptr(ng) = 0._r8
+               if (Sg_icemask_coupled_fluxes(ns)%ptr(ng).gt.0.d0) then
+                  if(Flgl_qice(ns)%ptr(ng) > 0.d0) then
+                     loc_pos_smb(1) = loc_pos_smb(1)+Flgl_qice(ns)%ptr(ng)*Sg_area(ns)%ptr(ng)
+                  end if
+                  ! Ignore places that are exactly 0.d0
+                  if(Flgl_qice(ns)%ptr(ng) < 0.d0) then
+                     loc_neg_smb(1) = loc_neg_smb(1)+Flgl_qice(ns)%ptr(ng)*Sg_area(ns)%ptr(ng)
+                  end if
                end if
             end do
+            ! Now do two global sums to get the ice sheet total positive
+            ! and negative ice fluxes
+            call ESMF_GridCompGet(gcomp, vm=vm, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            call ESMF_VMAllreduce(vm, senddata=loc_pos_smb, recvdata=Tot_pos_smb, count=1, &
+                 reduceflag=ESMF_REDUCE_SUM, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            call ESMF_VMAllreduce(vm, senddata=loc_neg_smb, recvdata=Tot_neg_smb, count=1, &
+                 reduceflag=ESMF_REDUCE_SUM, rc=rc)
+            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            
+            ! If there's more positive than negative, then set all
+            ! negative to zero and destribute the negative flux amount
+            ! across the positive values, scaled by the size of the
+            ! positive value. This section also applies to any chunks
+            ! where there is no negative smb. In that case the ice
+            ! runoff is exactly equal to the input smb.
+            if(abs(Tot_pos_smb(1)) >= abs(Tot_neg_smb(1))) then
+               do ng = 1,lsize             
+                  if (Sg_icemask_coupled_fluxes(ns)%ptr(ng).gt.0.d0) then
+                     if(Flgl_qice(ns)%ptr(ng) > 0.d0) then
+                        rat = Flgl_qice(ns)%ptr(ng)*Sg_area(ns)%ptr(ng)/Tot_pos_smb(1)
```

Comment:
As noted in our email discussion, I question the use of Sg_area here and on line 456.

Reply:
@Katetc - thanks for fixing the use of areas. I'm marking this comment as resolved.